model/src/util.py

```diff
@@ -44,48 +80,65 @@ def apply(self, error: float, dt: float = 1) -> float:
 
 class SpikeMovingAverage:
 
-    def __init__(self, tau_mean: float) -> None:
+    def __init__(self, tau_mean: float = 600) -> None:
         """
         tau_mean:
          * A time constant that determines the smoothing factor for the moving average
            of spikes.
          * A smaller tau_mean will make the average more sensitive to recent spikes.
          * A larger tau_mean will give a smoother average that is less responsive to
            individual spikes, reflecting a longer-term average rate.
+
+        Zenke's paper uses a tau_mean of 600s
         """
-        self.mean: float = 0
+        self.mean: Optional[torch.Tensor] = None
         self.tau_mean = tau_mean
 
-    def apply(self, spike: float, dt: float = 1) -> float:
+    def apply(self, spike: torch.Tensor, dt: float = 1) -> torch.Tensor:
+        if self.mean is None:
+            # Initialize mean based on the first spike received
+            self.mean = torch.zeros_like(spike)
```

Comment:
I'm slightly concerned about the weird transients that will result from the mismatch between this initialization and the the steady state mean. I guess a solution is to let the network run a settling period but the nature of the online rule is that this transient will be reflected on the network itself. Might be worth discussing how worried we should be about this, but this is not intended to a derailing comment simply one to ask: is there a way around this or is this something we have to live with?

Reply:
Okay yeah I will add a TODO to note and address this in the code.

Reply:
In a way this is sort of similar to initializing the activations in the FF network. In that specific case, initializing to zeros did the best. But likewise we will probably need to toy with this.